### ISSUE TITLE
Splash Page appears for all Launches but first. We have license.

### ISSUE THREAD

**Issue by Work1SaurGitHub** [Gamua/Adobe-Runtime-Support#455]
I have a bundle I made on Windows 10. I installed the license key and I can see a license.txt file in the \META-INF\AIR directory.

The first time I run the executable the splash page does not appear. On all subsequent runs the splash page appears. 

I am using Harman AIR version: 33.1.1.217. I am using the Adobe 4.6.0 SDK.

I think i have located part of the problem. The  id tag value in  META-INF/AIR application.xml increments each time it is launched.

I think I may have solved this. I am going to mark it closed now.

**Comment by Work1SaurGitHub** [comment]
I think I solved it by changing the format of the AIR ID.

**Comment by Work1SaurGitHub** [comment]
Hi Andrew,

Will do. I am very busy at the moment

I’ll get back to you later today or tomorrow.

**Comment by Work1SaurGitHub** [comment]
Hi 

The problem may be a peculiarity of our setup. Whenever we run the application it appends a number onto the  id tag in the META-INF/AIR application.xml i


![image](https://user-images.githubusercontent.com/46460340/92674335-ee667480-f2d1-11ea-84b2-72d3c469945f.png

This is an example

**Comment by ajwfrost** [comment]
Ah - working around the fact you can't otherwise have multiple instances. We'd been discussing with Adobe about removing this restriction, for installed apps anyway. But for now .. is there a pattern to your application ID modifications that we can detect i.e. do you have "-1", "-2" on the end? We could perhaps discard any numbers found after a dash/hyphen character, to catch this sort of scenario..

**Comment by Work1SaurGitHub** [comment]
If you could discard any numbers you found after the - hyphen that would work very well for us. We could easily make our application ID of the form RmsClient-6.

**Comment by ajwfrost** [comment]
Hi - we looked into this a little, part of the issue is going to be about shared resources and the multiple processes trying to access the same files at the same time e.g. for LSOs, ELS, named pipes (LocalConnection) etc. 

If you need multiple applications to be able to both have access to the same things, we would find this a little trickier to work around... but I will add this to the list for someone to make the initial changes, and we can get QA to then see what impacts there might be..

thanks

**Comment by Work1SaurGitHub** [comment]
Hi 
Thanks for the response.
Is there anyway you can just have the splash page/license check just ignore numbers after the dash.
For example:.  MyId-2, MyId-3, MyId-4 etc. would all be considered licensed if the folder was built with MyId-.  
We have been using the work around (repeating the SO link : https://stackoverflow.com/questions/2217307/starting-an-adobe-air-application-multiple-times)_ for years and it has been working great.  

The application is for internal users only and they all run on Windows 10.

I appreciate you cant just accommodate one client and allowing multiple applications at once could be substantial. What is blocking us is this splash page. Our user's will not like it. There is also concern at our end about risk. If you ever decide to get stricter about licenses and actually shut down rather than just show a splash page, that would create difficulties.  

So I am just wondering (don't know your code base and have maybe  3 months on Flex) - if altering the license key recognition behavior would be easier. That would resolve our difficulty.

**Comment by ajwfrost** [comment]
Hi - this is a good short-term compromise! Thanks for the suggestion. We'll still be looking at multiple apps; it just struck me that if you're using this workaround, then it already means that your different instances would not be able to share data stores etc if these are based on application IDs..? So a simple solution might also be for us to just build this workaround in as an official feature, e.g. if you added a <multipleLaunch>true</multipleLaunch> flag into the descriptor file, we would automatically just add a number to the app ID and allow subsequent versions to load...

We're just going through QA on a release candidate at the moment for an SDK update, but we'll be pushing out a new SDK fairly shortly after this one as there are iOS issues that we need to sort out! so it wouldn't be too long before we can get this into the release..

thanks

**Comment by ajwfrost** [comment]
FYI in case you want to try this, we're making available a patch that has another fix in it but also will cope with the -1, -2 pattern:
https://transfer.harman.com/messages/sN8r93jkimvNhr0IslQxXP/attachments/HefTUKNKVRnc3bxudQauxF/download/Adobe%20AIR.dll
Checksum (SHA256): 8454acbb53ac5ae69e134ea4b71944040727061f2dfe63333cfcadf7a35b7503

**Comment by Work1SaurGitHub** [comment]
Hi - I tried this out and it is working for us. Thank you. I notice though that the dll does not match the latest release. Would I be correct in expecting it to be added to a release soon.

**Comment by ajwfrost** [comment]
Yes we'll be doing another release soon to incorporate the other changes.
Curious though, what you do mean by 'does not match' the latest release?

**Comment by Work1SaurGitHub** [comment]
I did a binary compare on the .dll in the link you provided to the .dll files in the latest release on your website and they weren't the same. That is what I meant by did not match.
